### ISSUE TITLE
Add pragmas for compiler options

### ISSUE THREAD

**Issue by mkruselj** [nojanath/SublimeKSP#240]
We should be able to override the SublimeKSP settings with these pragma directives:

```
{ #pragma remove_whitespace true/false }
{ #pragma compact_variables true/false }
{ #pragma extra_syntax_checks true/false }
{ #pragma optimize_compiled_code true/false }
{ #pragma add_compile_date_comment true/false }
{ #pragma optimize_compiled_code true/false }
```

If `optimize_compiled_code`was set to true but `extra_syntax_checks` was not set to true, or the SublimeKSP native option is false, compiler should error out saying that extra syntax checks must be enabled for that option to work.

**Comment by JackWilliams-FractureSounds** [comment]
Add option to include multiple `save_compile_source`

**Comment by gabrieldib** [comment]
Jack already said that, but just to reinforce (and paste from #275)
please add the possibility to add multiple save_compile_source
this way we can have the compiled going into the /Resources/scripts folder AND a local copy inside the sources folder, to avoid having to copy the compiled code every time we need to check something.

also have the option to preserve the spaces / tabs. Quite useful if you need to read the compiled code.

**Comment by gabrieldib** [comment]
Another one, something like:
`{ #pragma init_ksp ./my_main_file.ksp`
this pragma would go at the top of all ksp files pointing to the file with the ICB
this way the user can press compile from anywhere and get the code compiled correctly, even if more than one project is open in that sublime window. There would be no need to find the tab with the main file (with the ICB) to press compile.

**Comment by mkruselj** [comment]
> to avoid having to copy the compiled code every time we need to check something.

You **do** know you can use symbolic links for this, right? :)

**Comment by gabrieldib** [comment]
I do use sym links already, it would be cool to not have to set them up for every project.

On the compile from any of the external KSP files topic, you mean that having the pragma in the main file would indicate that the plugin would read all the import statements of the main and external files and keep a cache to know if the user presses compile from any of the appointed files? Sounds like it could work.

**Comment by mkruselj** [comment]
> On the compile from any of the external KSP files topic, you mean that having the pragma in the main file would indicate that the plugin would read all the import statements of the main and external files and keep a cache to know if the user presses compile from any of the appointed files? Sounds like it could work and yes, prevents the user from having to put a pragma line on every file.

Yes, that is exactly my point. It's simply a way better design, that's also easier to use.
